Jetpack: Update Pricing Page with new Design and Pricing

cbauerman opened the pull request:
#### Changes proposed in this Pull Request

* Implement design changes from Simplify Pricing Project ( p1HpG7-dcK-p2 ) to the Jetpack Pricing Page

**DO NOT MERGE:** This PR references pricing that is not implemented yet.

**Monthly Selected ( before -> after )**

<img src="https://user-images.githubusercontent.com/2810519/138504415-03784043-7913-476e-9e59-6f4323ad3cdb.png" width=350 align="left">
<img src="https://user-images.githubusercontent.com/2810519/138503569-ceec7eab-0fd7-430a-b5e2-3286798618f2.png" width=350 >
<br />

**Yearly Selected ( before -> after )**
<img src="https://user-images.githubusercontent.com/2810519/138504399-aabd357d-b115-4956-a348-86ccbcecfcef.png" width=350 align="left">
<img src="https://user-images.githubusercontent.com/2810519/138503694-19d16c10-be8d-49e0-aeed-4589cb506e1b.png" width=350 >
<br />

#### Testing instructions


1. Boot branch or use live branch and navigate to `/pricing` with Jetpack Cloud
2. Check that the design matches the above "after" screenshots

cbauerman committed:
Update copy to "billed yearly"

cbauerman committed:
revert changes to coupon injection

a8ci18n commented on client/components/jetpack/card/jetpack-product-card/display-price/paid.tsx:
```diff
@@ -43,19 +43,20 @@ const Paid: React.FC< OwnProps > = ( {
 	const discountElt =
 		billingTerm === TERM_ANNUALLY
-			? translate( 'Save %(percent)d%% for the first year ✢', {
+			? translate( 'Save %(percent)d%% for the first year *', {
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented on client/components/jetpack/card/jetpack-product-card/display-price/time-frame.tsx:
```diff
@@ -18,10 +18,10 @@ const TimeFrame: React.FC< TimeFrameProps > = ( { expiryDate, billingTerm } ) =>
 	const billingTermText = useMemo( () => {
 		if ( billingTerm === TERM_MONTHLY ) {
-			return translate( '/month, paid monthly' );
+			return translate( '/month, billed monthly' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'per month, billed monthly' )` *ES Score: 13*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/simplify-jetpack-pricing-page)

a8ci18n commented on client/components/jetpack/card/jetpack-product-card/display-price/time-frame.tsx:
```diff
@@ -18,10 +18,10 @@ const TimeFrame: React.FC< TimeFrameProps > = ( { expiryDate, billingTerm } ) =>
 		}
 
-		return translate( '/month, paid yearly' );
+		return translate( '/month, billed yearly' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **16** times:
`translate( 'per month, billed yearly', { context: 'logged-out-homepage'} )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/simplify-jetpack-pricing-page)

a8ci18n commented on client/components/jetpack/intro-pricing-banner/index.tsx:
```diff
@@ -12,14 +12,19 @@ const Banner: React.FC = () => {
 				<p className="intro-pricing-banner__call-to-action">
 					{ translate(
-						'Get the perfect Jetpack for your site with %(percent)d%% off the first term. Try it risk free with our %(days)d-day money-back guarantee.*',
+						'Get the perfect Jetpack for your site with %(percent)d%% off the first year. Try it risk free with our %(days)d-day money-back guarantee.**',
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented on client/my-sites/plans/jetpack-plans/footnotes-list/index.tsx:
```diff
@@ -2,16 +2,19 @@ import { useTranslate } from 'i18n-calypso';
 			</li>
 			<li className="footnotes-list__item">
-				{ translate( '✢ Discount is for first term only, all renewals are at full price.' ) }
+				{ translate( '** Monthly plans are 7-day money back guarantee.', { comment: '** ' } ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '* Monthly plans are 7-day money back guarantee.' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/simplify-jetpack-pricing-page)

cbauerman committed:
Remove "Most Popular" title

cbauerman committed:
Refine Plan Filter bar to center on toggle

cbauerman committed:
Translation comments, removed unused style lines

a8ci18n commented on client/my-sites/plans/jetpack-plans/footnotes-list/index.tsx:
```diff
@@ -2,16 +2,21 @@ import { useTranslate } from 'i18n-calypso';
 			</li>
 			<li className="footnotes-list__item">
-				{ translate( '✢ Discount is for first term only, all renewals are at full price.' ) }
+				{ translate( '** Monthly plans are 7-day money back guarantee.', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '* Monthly plans are 7-day money back guarantee.' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/simplify-jetpack-pricing-page)

monsieur-z reviewed (changes requested):
Code looks good, and the page looks similar to the screenshots.

The toggle + discount seems unbalanced to me, but feel free to discard my comment if that's part of the original design.
<img width="467" alt="Screen Shot 2021-10-25 at 11 31 17 AM" src="https://user-images.githubusercontent.com/1620183/138726234-9c7f602b-00a4-4cfc-a22a-97e1dde9a98a.png">

However I think the toggle should be centered on mobile:


<img width="400" alt="Screen Shot 2021-10-25 at 11 34 51 AM" src="https://user-images.githubusercontent.com/1620183/138726431-594548e9-d343-468d-9d50-679f80a5c455.png">
On a slightly larger screen, the toggle + discount seems off as well:
<img width="498" alt="Screen Shot 2021-10-25 at 11 27 35 AM" src="https://user-images.githubusercontent.com/1620183/138726365-24e649b3-8b7d-4539-b13f-686ea0e85039.png">

cbauerman committed:
Restore filter bar style, use new breakpoints

monsieur-z reviewed (approved):
LGTM!

cbauerman committed:
Make Grid Section Titles Optional

- Remove the Property Requirement for `ProductGridSection`s to have a title
- Remove the "Most Popular" title when `jetpack/simplify-pricing-structure` is active

cbauerman committed:
Chunk of design refinements

a8ci18n commented on client/components/jetpack/intro-pricing-banner/index.tsx:
```diff
@@ -12,15 +12,20 @@ const Banner: React.FC = () => {
 					{ preventWidows(
 						translate(
-							'Get the perfect Jetpack for your site with %(percent)d%% off the first term. Try it risk free with our %(days)d-day money-back guarantee.*',
+							'Get the perfect Jetpack for your site with %(percent)d%% off the first year. Try it risk free with our %(days)d-day money-back guarantee.**',
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

cbauerman committed:
Remove slanted discount callout

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7031639

Thank you @cbauerman for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7031639).
